tests/test_10_base.py

```diff
@@ -15,66 +11,46 @@ class BaseNameTest(Archive):
     _mimes = ["mime1", "mime2"]
 
 
-class Archive(unittest.TestCase):
-    def test_10_guess_basename(self):
-        fileobj = BytesIO(b"")
-        try:
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("mime2", "xxx", fileobj)
-            )
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("mime1", "xxx.ext2", fileobj)
-            )
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("mime2", "xxx.ext1", fileobj)
-            )
-        except ValueError as e:
-            self.fail(repr(e))
-        try:
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("xxx", "xxx.ext1", fileobj)
-            )
-        except ValueError:
-            pass
-        else:
-            self.fail("guessing should has failed")
-
-
-class ArchiveFileTest(unittest.TestCase):
-    def _regular_tests(self, archive, fileobj, filename, text):
-        self.assertEqual(
-            archive.fileno(), fileobj.fileno(), "file no does not match!"
-        )
-        self.assertEqual(
-            archive.name, filename, "name attribute does not match!"
-        )
+def test_10_guess_basename():
+    fileobj = BytesIO(b"")
+    assert "xxx" == BaseNameTest._guess("mime2", "xxx", fileobj)
+    assert "xxx" == BaseNameTest._guess("mime1", "xxx.ext2", fileobj)
+    assert "xxx" == BaseNameTest._guess("mime2", "xxx.ext1", fileobj)
+    with pytest.raises(ValueError):
+        assert BaseNameTest._guess("xxx", "xxx.ext1", fileobj)
+
+
+def test_10_passing_file_object(tmp_path):
```

Comment:
`tmp_path` is a magic variable that I cannot figure out how it got there by looking at this diff.

This magic is both lovely and worrisome as a little bit is great but it's easy to become mysterious.

Reply:
Important question, thank you. That's one of the [`pytest`'s fixtures](https://docs.pytest.org/en/stable/tmpdir.html) that will create a directory structure in `/tmp/pytest-of-$USER` with the latest three test runs and a subdirectory for each test function that uses it. That's very helpful to be able to inspect the files after the test is over and it also removes the whole housekeeping from the test suite, such as removal of temporary files and directories.

And it is a `pathlib.Path` object, so exactly what we want. When this PR is merged, I'd like to change all strings in the project that represent directories or files to a `pathlib.Path` object anyway, and make `destream`'s API accept both.

Reply:
> This magic is both lovely and worrisome as a little bit is great but it's easy to become mysterious.

It is a very good point and a major concern when you design an API for a library: if you make it "too" clever, people won't like because they feel they lose control, you make it too stupid you might end up with a lot of boilerplate.

For me py(.)test is familiar so I have no issue with it. I prefer it over unittest because there is a lot of boilerplate in unittest. unittest feels like java stuff :grin: 

Well I don't have much to say except that you raised an important point about programming and designing APIs. In this particular case I would suggest you to get used to pytest because it is a really common test framework in Python. Good stuff to know

Reply:
> unittest feels like java stuff grin

with all that `assertEqual` CamelCase, yes.

`pytest` allows to write natural expressions in the `assert` lines, which is just better to read.

Reply:
@jruere here are they: https://docs.pytest.org/en/stable/fixture.html

I sometimes use `request`, which gives me in `request.node.name` the name of the test function itself, which is practical if you need a unique identifiable name for something.

Reply:
Fixtures seem nice!